modules/generator/processor/spanmetrics/spanmetrics.go

```diff
@@ -128,6 +133,130 @@ func (p *Processor) aggregateMetricsForSpan(svcName string, rs *v1.Resource, spa
 	p.spanMetricsDurationSeconds.ObserveWithExemplar(registryLabelValues, latencySeconds, tempo_util.TraceIDToHexString(span.TraceId), spanMultiplier)
 }
 
+// applyFilterPolicy returns true if the span should be included in the metrics.
+func (p *Processor) applyFilterPolicy(rs *v1.Resource, span *v1_trace.Span) bool {
+	// With no filter policies specified, all spans are included.
+	if len(p.Cfg.FilterPolicies) == 0 {
+		return true
+	}
+
+	for _, policy := range p.Cfg.FilterPolicies {
+		if policy.Include != nil {
+			if !policyMatch(policy.Include, rs, span) {
+				return false
+			}
+		}
+
+		if policy.Exclude != nil {
+			if policyMatch(policy.Exclude, rs, span) {
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
+// policyMatch returns true when the resource attribtues and span attributes match the policy.
+func policyMatch(policy *PolicyMatch, rs *v1.Resource, span *v1_trace.Span) bool {
```

Comment:
I tried it that way also, but I found it less readable.  Since we're passing a pointer here it should be cheap, but I think if we want to construct the three policies to match outside of this method then we'll pass all thee in, which might clean up this method, but muddy the water on the other.  Are you thinking it might be more readable if we move it up a layer?

Reply:
Sorry yup that's what I meant, just the three policies since they are static

Reply:
I'll commit an update here shortly with your suggestion.  Here are the benchmark results.
```
❯ benchstat old.txt new.txt
goos: linux
goarch: amd64
pkg: github.com/grafana/tempo/modules/generator/processor/spanmetrics
cpu: Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz
                                      │   old.txt    │                new.txt                │
                                      │    sec/op    │    sec/op     vs base                 │
SpanMetrics_applyFilterPolicyNone-8      32.31 ± ∞ ¹    33.50 ± ∞ ¹        ~ (p=1.000 n=1) ²
SpanMetrics_applyFilterPolicySmall-8    220.3m ± ∞ ¹   128.1m ± ∞ ¹        ~ (p=1.000 n=1) ²
SpanMetrics_applyFilterPolicyMedium-8   871.6m ± ∞ ¹   216.6m ± ∞ ¹        ~ (p=1.000 n=1) ²
geomean                                  1.837         976.0m        -46.88%
¹ need >= 6 samples for confidence interval at level 0.95
² need >= 4 samples to detect a difference at alpha level 0.05

                                      │      old.txt       │                new.txt                 │
                                      │        B/op        │     B/op       vs base                 │
SpanMetrics_applyFilterPolicyNone-8          15.67Gi ± ∞ ¹   15.67Gi ± ∞ ¹        ~ (p=1.000 n=1) ²
SpanMetrics_applyFilterPolicySmall-8     32001685.00 ± ∞ ¹     80.00 ± ∞ ¹        ~ (p=1.000 n=1) ²
SpanMetrics_applyFilterPolicyMedium-8   224008272.00 ± ∞ ¹     80.00 ± ∞ ¹        ~ (p=1.000 n=1) ²
geomean                                      471.2Mi         46.46Ki        -99.99%
¹ need >= 6 samples for confidence interval at level 0.95
² need >= 4 samples to detect a difference at alpha level 0.05

                                      │      old.txt      │                new.txt                │
                                      │     allocs/op     │  allocs/op    vs base                 │
SpanMetrics_applyFilterPolicyNone-8          170.9M ± ∞ ¹   170.9M ± ∞ ¹        ~ (p=1.000 n=1) ²
SpanMetrics_applyFilterPolicySmall-8    1000004.000 ± ∞ ¹    3.000 ± ∞ ¹        ~ (p=1.000 n=1) ²
SpanMetrics_applyFilterPolicyMedium-8   3000004.000 ± ∞ ¹    3.000 ± ∞ ¹        ~ (p=1.000 n=1) ²
geomean                                      8.004M         1.154k        -99.99%
¹ need >= 6 samples for confidence interval at level 0.95
² need >= 4 samples to detect a difference at alpha level 0.05
```